core/deployment/src/main/java/io/quarkus/deployment/util/DeploymentUtil.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.deployment.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+public class DeploymentUtil {
+
+    public static final String DEPLOY = "quarkus.%s.deploy";
+    private static final Pattern QUARKUS_DEPLOY_PATTERN = Pattern.compile("quarkus\\.([^\\.]+)\\.deploy");
+
+    /**
+     * Get the available deployers.
+     * The list is obtained by checking for properties `quarkus.xxx.deploy`.
+     * These properties have a default value and thus they will be found regardless of the
+     * actual user configuration, so we check for property names instead.
+     *
+     * @return a {@link List} with all available deployers.
+     */
+    public static List<String> getDeployers() {
+        Config config = ConfigProvider.getConfig();
+        return Stream.concat(System.getProperties().entrySet().stream().map(e -> String.valueOf(e.getKey())),
+                StreamSupport.stream(config.getPropertyNames().spliterator(),
```

Comment:
Shouldn't `config` already include `System.getProperties()`?

Reply:
I think that this was not that case, but if you are positive I can have a second look.

Reply:
Removed the System property check and retried all the scenarios. It seems to be working even without it. Updated the PR accordingly.